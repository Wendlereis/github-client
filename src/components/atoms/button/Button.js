import React from 'react'
import PropTypes from 'prop-types'
import styled, { css } from 'styled-components'

const StyledButton = styled.button`
  padding: 10px 15px;
  border: 0;
  border-radius: 3px;
  background-color: #f0f0f0;
  color: #52bdb2;
  text-transform: uppercase;
  font-weight: 500;

  &:focus {
    outline: none;
  }

  &:hover {
    background-color: #e0e0e0;
    transition: all 0.5s;
  }

  ${props => {
    return (
      props.primary &&
      css`
        background: #ff5a49;
        color: #ffffff;

        &:hover {
          background-color: #ff2c16;
        }
      `
    )
  }}
`
export default function Button(props) {
  return <StyledButton {...props} />
}

Button.propTypes = {
  primary: PropTypes.bool
}
