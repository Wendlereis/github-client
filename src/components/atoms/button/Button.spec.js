import React from 'react'
import { shallow } from 'enzyme'
import Button from './Button'

describe('Button', () => {
  it('should render a default button', () => {
    const component = shallow(<Button>This is a default button</Button>)

    expect(component).toMatchSnapshot()
  })

  it('should render a primary button', () => {
    const component = shallow(<Button primary>This is a primary button</Button>)

    expect(component).toMatchSnapshot()
  })
})
