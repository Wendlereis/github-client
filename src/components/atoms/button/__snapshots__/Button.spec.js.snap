// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Button should render a default button 1`] = `
<styled.button>
  This is a default button
</styled.button>
`;

exports[`Button should render a primary button 1`] = `
<styled.button
  primary={true}
>
  This is a primary button
</styled.button>
`;
